src/main/java/com/exasol/projectkeeper/ProjectFilesFitter.java

```diff
@@ -0,0 +1,75 @@
+package com.exasol.projectkeeper;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.StandardCopyOption;
+import java.util.List;
+
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugin.logging.Log;
+
+/**
+ * This class shapes the project's file structure.
+ */
+public class ProjectFilesFitter extends AbstractProjectFilesWalker {
+    private final Log log;
+
+    /**
+     * Crate a new instance of {@link ProjectFilesFitter}.
+     * 
+     * @param log logger
+     */
+    public ProjectFilesFitter(final Log log) {
+        super();
+        this.log = log;
+    }
+
+    /**
+     * Fit the file structure of the project.
+     * 
+     * @param projectDirectory project's base directory
+     * @param enabledModules   list of enabled modules
+     * @throws MojoFailureException in case fitting failed
+     */
+    public void fitProjectStructure(final File projectDirectory, final List<String> enabledModules)
+            throws MojoFailureException {
+        final FitterTemplateVisitor fitterVisitor = new FitterTemplateVisitor(this.log);
+        super.run(fitterVisitor, projectDirectory, enabledModules);
+        if (fitterVisitor.hadErrors()) {
+            throw new MojoFailureException("E-PK-5 Failed to fit repository file structure. See previous errors.");
```

Comment:
I suggest to change the `fit` to `create`.
"Failed to create repository file structure" is much clearer.